### PR TITLE
Change plugin dev command `make debug-plugin` to `make attach-headless`

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

The make command `make attach-headless` was originally called `make debug-plugin` in earlier stages of plugins (such as the Jira plugin). I originally thought that's what it is called in the starter template, but I don't think the starter template ever had the `debug-plugin` command, so I propose we just replace the name outright in the docs.

### CODE REVIEW

**Comment by catalintomai** [comment]
> make attach-headless

@mickmister , in which Makefile file is the "attach-headless"? I was using the "debug-server-headless", is that the one you are referring to?

**Comment by mickmister** [comment]
@catalintomai These docs are for plugin development. The `Makefile` in every plugin repo should have the `make attach-headless` command, if it was created from the starter template. Running the command attaches to a live plugin process, and lets you debug the plugin's server code from an external tool, just as `make debug-server-headless` does for the MM server code.

**Comment by catalintomai** [comment]
Thanks, wonder why we have a different name since the intent is the same (or call it instead smth. like `debug-plugin-headless`).

**Review by hanzei — APPROVED** [review]

**Comment by mickmister** [comment]
> @mickmister, thanks, wonder why we have a different name since the intent is the same (or call it instead smth. like `debug-plugin-headless`).

@catalintomai Good point. It should be named `debug-plugin-headless`, but `attach-headless` has been in the starter template for a while, so many repos have the command already.

**Review by jfrerich — APPROVED** [review]

File: site/content/extend/plugins/developer-workflow.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ subsection: Plugins
 - `make enable` - Enables the plugin on the Mattermost server
```
Comment (jfrerich) [inline site/content/extend/plugins/developer-workflow.md:15]: interesting.  I haven't seen this command.

Comment (mickmister) [reply site/content/extend/plugins/developer-workflow.md:15]: @jfrerich It should be available for all repos sync'd with the starter template. `make reset` is usually more useful though, which is like running `make disable && make enable`

File: site/content/extend/plugins/developer-workflow.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ subsection: Plugins
...
 - `make disable` - Disables the plugin on the Mattermost server
```
Comment (justinegeffen) [inline site/content/extend/plugins/developer-workflow.md:16]:
```suggestion
- `make disable` - Disables the plugin on the Mattermost server.
```

File: site/content/extend/plugins/developer-workflow.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ subsection: Plugins
...
 - `make reset` - Disables and re-enables the plugin on the Mattermost server
```
Comment (justinegeffen) [inline site/content/extend/plugins/developer-workflow.md:17]:
```suggestion
- `make reset` - Disables and re-enables the plugin on the Mattermost server.
```

File: site/content/extend/plugins/developer-workflow.md [hunk 4]
```diff
@@ -15,7 +15,7 @@ subsection: Plugins
...
-- `make debug-plugin` - Starts a `delve` process and attaches it to your running plugin
+- `make attach-headless` - Starts a `delve` process and attaches it to your running plugin
```
Comment (justinegeffen) [inline site/content/extend/plugins/developer-workflow.md:18]:
```suggestion
- `make attach-headless` - Starts a `delve` process and attaches it to your running plugin.
```

File: site/content/extend/plugins/developer-workflow.md [hunk 5]
```diff
@@ -51,7 +51,7 @@ An `http` URL pointing to your server should show in the terminal. The `https` v
 
 ### Debugging server-side plugins using `delve`
 
-Using the `make debug-plugin` command will allow you to use a debugger and step through the plugin's server code. A [delve](https://github.com/go-delve/delve) process will be created and attach to your plugin. You can then use an IDE or debug console to connect to the `delve` process. If you are using VSCode, you can use this `launch.json` configuration to connect.
+Using the `make attach-headless` command will allow you to use a debugger and step through the plugin's server code. A [delve](https://github.com/go-delve/delve) process will be created and attach to your plugin. You can then use an IDE or debug console to connect to the `delve` process. If you are using VSCode, you can use this `launch.json` configuration to connect.
```
Comment (justinegeffen) [inline site/content/extend/plugins/developer-workflow.md:54]:
```suggestion
Using the `make attach-headless` command will allow you to use a debugger and step through the plugin's server code. A [delve](https://github.com/go-delve/delve) process will be created and attach to your plugin. You can then use an IDE or debug console to connect to the `delve` process. If you're using VSCode, you can use this `launch.json` configuration to connect.
```

**Review by justinegeffen — APPROVED** [review]

**Pull request merged**